tests/Integration/ORM/EntityRelationship/EntityFactoryRelationshipTestCase.php

```diff
@@ -519,6 +522,137 @@ public function can_call_create_in_after_persist_callback(): void
         self::assertSame(unproxy($category), $category->getContacts()[0]?->getCategory());
     }
 
+    /** @test */
+    #[Test]
+    #[DataProvider('provideCascadeRelationshipsCombinations')]
+    #[UsingRelationships(Contact::class, ['address'])]
+    public function can_use_nested_after_persist_callback(): void
+    {
+        $contact = static::contactFactory()::createOne(
+            [
+                'address' => static::addressFactory()
+                    ->afterPersist(function(Address $address) {
+                        $address->setCity('city from after persist');
+                    }),
+            ]
+        );
+
+        self::assertSame('city from after persist', $contact->getAddress()->getCity());
+    }
+
+    /** @test */
+    #[Test]
+    #[DataProvider('provideCascadeRelationshipsCombinations')]
```

Comment:
Don't we need an `@dataProvider` annotation also?

Reply:
nope, the whole mechanism used to provide combinations for cascade relations does not support phpunit 9